### PR TITLE
Otlp http receiver transport

### PR DESCRIPTION
**Pull request by jonathonfletcher**
**Description:**

Enable protocols.http.transport config, comparable to protocols.grpc.transport config.

Useful in mixed ipv4 / ipv6 environments where the `"tcp"` transport does not bind as desired. enables use of transport `"tcp4"` and transport `"tcp6"` in http receiver equivalent to same in gprc receiver.

**Testing:**

Tested with:
- `transport: tcp4`
- `transport: tcp6`
- `transport: tcp`

to confirm same bind as grpc receiver with same config setting.Are there gprc receiver transport configure tests in-tree - the existing tests all seem to wire to "tcp".

**Documentation:** 

Added line in `receiver/otlpreceiver/config.md` under `confighttp-HTTPServerSettings` with `transport` equivalent to same under `configgrpc-GRPCServerSettings`.

### CODE REVIEW

File: config/confighttp/confighttp.go [hunk 1]
```diff
@@ -203,6 +203,9 @@ type HTTPServerSettings struct {
 	// Endpoint configures the listening address for the server.
 	Endpoint string `mapstructure:"endpoint"`
```
Comment (bogdandrutu) [inline config/confighttp/confighttp.go:204]: If needed to extend with transport, we should "embed" (using mapstructure `squash`) https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confignet/confignet.go#L22 into this config.

File: config/confighttp/confighttp.go [hunk 2]
```diff
@@ -203,6 +203,9 @@ type HTTPServerSettings struct {
...
 
+	// Server Transport config. For transport only "tcp" / "tcp4" / "tcp6" are valid options.
+	Transport string `mapstructure:"transport"`
```
Comment (bogdandrutu) [inline config/confighttp/confighttp.go:207]: We need unit tests for this.

Comment (bogdandrutu) [reply config/confighttp/confighttp.go:207]: Is this comment correctly place in this thread? I don't feel that a unittest will break anything.

Comment (jpkrohling) [reply config/confighttp/confighttp.go:207]: This is not a unit test O_O

Comment (jpkrohling) [reply config/confighttp/confighttp.go:207]: Sorry, just realized I hijacked your comment. My comments are a different concern than yours.

**Pull request closed without merging**